sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/controller/group/ThroughputGroupControllerBase.java

```diff
@@ -0,0 +1,238 @@
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkArgument;
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+/**
+ * Throughput group controller. Two common tasks across all group controller implementations:
+ * 1. Create and initialize request controller based on connection mode
+ * 2. Schedule reset throughput uage every 1s.
+ */
+public abstract class ThroughputGroupControllerBase implements IThroughputController {
+    private final static Logger logger = LoggerFactory.getLogger(ThroughputGroupControllerBase.class);
+    private final Duration DEFAULT_THROUGHPUT_USAGE_RESET_DURATION = Duration.ofSeconds(1);
+
+    private final ConnectionMode connectionMode;
+    private final GlobalEndpointManager globalEndpointManager;
+    private final ThroughputControlGroup group;
+    private final AtomicReference<Double> groupThroughput;
+    private final AtomicInteger maxContainerThroughput;
+    private final RxPartitionKeyRangeCache partitionKeyRangeCache;
+    private final AsyncCache<String, IThroughputRequestController> requestControllerAsyncCache;
+    private final String targetContainerRid;
+
+    private final CancellationTokenSource cancellationTokenSource;
+    private final Scheduler scheduler;
+
+    public ThroughputGroupControllerBase(
+        ConnectionMode connectionMode,
+        GlobalEndpointManager globalEndpointManager,
+        ThroughputControlGroup group,
+        Integer maxContainerThroughput,
+        RxPartitionKeyRangeCache partitionKeyRangeCache,
+        String targetContainerRid) {
+
+        checkNotNull(globalEndpointManager, "Global endpoint manager can not be null");
+        checkNotNull(group, "Throughput control group can not be null");
+        checkNotNull(partitionKeyRangeCache, "Partition key range cache can not be null or empty");
+        checkArgument(StringUtils.isNotEmpty(targetContainerRid), "Target container rid cannot be null nor empty");
+
+        this.connectionMode = connectionMode;
+        this.globalEndpointManager = globalEndpointManager;
+        this.group = group;
+
+        if (this.group.getTargetThroughputThreshold() != null) {
+            checkNotNull(maxContainerThroughput, "Max container throughput can not be null when target throughput threshold defined");
+            this.maxContainerThroughput = new AtomicInteger(maxContainerThroughput);
+        } else {
+            this.maxContainerThroughput = null;
+        }
+
+        this.groupThroughput = new AtomicReference<>();
+        this.calculateGroupThroughput();// Calculate after maxContainerThroughput
+
+        this.partitionKeyRangeCache = partitionKeyRangeCache;
+        this.requestControllerAsyncCache = new AsyncCache<>();
+        this.targetContainerRid = targetContainerRid;
+
+        this.cancellationTokenSource = new CancellationTokenSource();
+        this.scheduler = Schedulers.elastic();
```

Comment:
I think we should use `Schedulers.parallel()` here, also, why capture the instance in `schedulers` instance field ?
Reactor `Schedulers` already cache their schedulers, so need to capture the instance here. Just use `Schedulers.elastic()` or `Schedulers.parallel()` everywhere in this class.